### ISSUE TITLE
Enhance support for Checksums

### ISSUE THREAD

**Issue by theonlypwner** [github/linguist#5935]
## Describe the enhancement

Continuation of #5130, #5138

- [ ] Support cksum format `checksum size` and `checksum size name`
- [ ] Add `.digest` extension
- [ ] Support popular formats in `.digest` files

**Comment by theonlypwner** [comment]
@Alhadis, this is from #5130:

>>> but is there a way to highlight it when it is explicitly specified, such as ```hash blocks in Markdown?
>>
>> Unfortunately not. ```hash blocks are highlighted by the same mechanism responsible for highlighting a file's source-code. For all intents and purposes, they're exactly the same.
>
>Don't the blocks below highlight differently? They are ` ```c `, ` ```html `, and ` ``` ` blocks, respectively.
>
>```c
>int main() {} // <strong attr="val">HTML</strong>
>```
>
>```html
>int main() {} // <strong attr="val">HTML</strong>
>```
>
>```
>int main() {} // <strong attr="val">HTML</strong>
>```

Here's what I was trying to get at with that post:

Since the highlighting can be controlled by the [info string](https://github.github.com/gfm/#info-string) after the three backticks, how does that work? Is it related to the `aliases` property? If so, can we use that to let users specify `cksum` specifically? Or is `cksum` not even popular enough for us to care about?

**Comment by Nixinova** [comment]
> Since the highlighting can be controlled by the [info string](https://github.github.com/gfm/#info-string) after the three backticks, how does that work? Is it related to the `aliases` property?

Yes, if the content of that infostring matches the language name or any of its aliases then that language is applied.

**Comment by Alhadis** [comment]
@theonlypwner Any of the aliases listed in [`languages.yml`](https://github.com/github/linguist/blob/d7799da826e01acdb8f84694d33116dccaabe9c2/lib/linguist/languages.yml#L964-L969) will activate checksum highlighting. Be aware that there's no correlation between which alias you pick and what checksums get highlighted; i.e., _all_ supported checksum patterns will be highlighted, even if the code-block was tagged with, say, ` ```md5 ` (a hypothetical example, seeing as how `md5` isn't currently supported as an alias). 

> If so, can we use that to let users specify `cksum` specifically? Or is `cksum` not even popular enough for us to care about?

I think we can add `cksum` as an alias. As a file-extension, a [quick search for `*.cksum` files](https://cs.github.com/?scopeName=All+repos&scope=&q=path%3A**%2F*.cksum+NOT+nothack) shows there's some usage on GitHub. You'll need to submit a pull-request and wait for @lildude to run an evaluation of its usage to know for sure, though.

**Comment by theonlypwner** [comment]
@Alhadis My point was that if we create another language entry with `aliases`, `filenames`, and `extensions` specifically for `cksum`, we can highlight in the `cksum` format for only ` ```cksum` blocks and `*.cksum` files because [you said it's better to have _no_ highlighting than wrong highlighting](https://github.com/github/linguist/issues/5130#issuecomment-758395847). Creating another language entry would create the correlation between the chosen alias and how the highlighting is performed.

But if the `cksum` format isn't popular enough, then it might not be necessary. In a [search for `*.cksum` files](https://github.com/search?q=extension%3Acksum&type=Code), most results look like the `SHA1SUMS` file format, which is already supported. I would recommend adding `.cksum` to `extensions`, which I have added to the task list. I don't think it is necessary to add `cksum` to `aliases`.

---

I will also respond to [your other comment in #5138](https://github.com/github/linguist/pull/5138#issuecomment-1159393206) here:

> Well, I can add support for patterns [like this](https://cs.github.com/geosolutions-it/ogc-testbed14-styles/blob/03d9b77d22953308eee88743f34ac4dc3e13ef0b/security/masterpw.digest?q=path%3A*.digest+NOT+nothack#L1)…
>
> ~~~
> digest1:dwLYXWqDkzaJhlo1P6sQqpxgU5cBRRYM9Sna/buqhHIoAX6/np9fYp9Zbgb3Rc7I
> ~~~
>
> … however, I'm unfamiliar with whatever program generates this kind of output, and I'd ideally like some authoritative reference on what formats are accepted in such lines (the example looks to be using base64, but I'd rather not go by guesswork if I can help it).

It seems like `digest1:` is an encryption prefix in [GeoServer passwd files](https://docs.geoserver.org/master/en/user/security/passwd.html). That prefix, `crypt1:`, and `crypt2:` precede a base64 string, while `plain:` precedes a plaintext password. Should passwd-type files also be included in the `Checksums` language, or should that be in another language like `passwd`?

> > Also, I noticed some files in the format `name whitespace hash` instead of `hash whitespace name`, which I hope either won't cause issues for the syntax highlighter or can be fixed later.
>
> There's no foolproof way to tell `name hash` from a filename that contains spaces, followed by a word that happens to contain <var>n</var> alphanumeric characters. Conversely, a line that begins with precisely <var>n</var> characters is a little less ambiguous. Consider files that begin with arbitrary text (e.g., [comments](https://github.com/ssdi-drive/nuxeo-drive/blob/8cc64b77cc464ff87cf9f8959d0a6f7085627375/tools/checksums.txt)), or even files [like this](https://github.com/molgryn/facebook-clang-plugins/blob/18bf6770420d4d53ce5bbc1bfbee909f3662c9d9/clang/src/checksums.txt#L4) that aren't technically a checksum list, but are still highlighted non-intrusively.

If there's no foolproof way, should we just simply give up and not support the `name hash` format?
